tools/nl/embeddings/utils.py

```diff
@@ -111,87 +116,95 @@ def build_and_save_preindex(fm: FileManager) -> List[SentenceObject]:
             text2sv[text] = set()
           text2sv[text].add(row[_COL_DCID])
 
-  sentences = [
-      SentenceObject(text, ';'.join(sorted(dcids)), [])
+  preindexes = [
+      PreIndex(text, ';'.join(sorted(dcids)))
       for text, dcids in text2sv.items()
   ]
-  sentences.sort(key=lambda x: x.text)
+  preindexes.sort(key=lambda x: x.text)
 
-  # Write preindex as CSV
+  # Write preindexes as CSV
   with open(fm.preindex_csv_path(), 'w') as csvfile:
     csv_writer = csv.writer(csvfile, delimiter=',')
     csv_writer.writerow([_COL_SENTENCE, _COL_DCID])
-    for sentence in sentences:
-      csv_writer.writerow([sentence.text, sentence.dcid])
+    for preindex in preindexes:
+      csv_writer.writerow([preindex.text, preindex.dcid])
 
-  # Write md5sum of preindex as a file
+  # Write md5sum of preindexes as a file
   with open(os.path.join(fm.local_output_dir(), _MD5_SUM_FILE), 'w') as f:
     f.write(get_md5sum(fm.preindex_csv_path()))
 
-  return sentences
+  return preindexes
 
 
-def retrieve_embeddings(
+def compute_embeddings(
     model: EmbeddingsModel,
-    target_sentences: List[SentenceObject],
-    saved_sentences: List[SentenceObject],
-) -> List[SentenceObject]:
-  """Compute embeddings for a list of sentence objects"""
+    preindexes: List[PreIndex],
+    existing_embeddings: List[Embedding],
+) -> List[Embedding]:
+  """Compute embeddings for the given preindexes
+
+  Args:
+    model: The embeddings model object,
+    preindexes: A list of preindex to compute embeddings for
+    existing_embeddings: A list of embeddings from previous run.
+  Return:
+    A list of embeddings for the preindexes.
+  """
   logging.info("Compute embeddings")
-  logging.info("Target sentences: %d", len(target_sentences))
+  logging.info("Preindex size: %d", len(preindexes))
   start = time.time()
-  # Find sentences that are not in saved_sentences
-  saved_text2sentence = {x.text: x for x in saved_sentences}
-  filtered_sentences = [
-      x for x in target_sentences if x.text not in saved_text2sentence
+  # Find preindex that are not in the existing embeddings
+  existing_embeddings_map = {x.preindex.text: x for x in existing_embeddings}
+  filtered_preindexes = [
+      x for x in preindexes if x.text not in existing_embeddings_map
   ]
-  logging.info("Filtered sentences to compute embeddings with model: %d",
-               len(filtered_sentences))
+  logging.info("Size of preindexes to compute embeddings with model: %d",
```

Comment:
nit:  For readability, instead of looping over preindexes twice, how about loop once and do:

```
# Prepare a list of preindexes for computation
result: list[Embedding] = []
preindexes_to_compute: list[Preindex] = []
for p in preindexes:
  if x.text in existing_embeddings_map:
    # Save computation, add to result
    result.append(...)
  else:
    # Compute embedding
    preindexes_to_compute.append(p)
```

Reply:
yes, this is better.